scipy/stats/stats.py

```diff
@@ -2089,29 +2102,99 @@ def percentileofscore(a, score, kind='rank'):
+
+    >>> stats.percentileofscore([1, 2, 3, 3, 4], [2, 3])
+    array([40., 70.])
+
+    The inputs can be infinite:
+
+    >>> stats.percentileofscore([-np.inf, 0, 1, np.inf], [1, 100])
+    array([75., 75.])
+
+    If `a` is empty, or containing only nan's (which are omitted),
+    the resulting percentiles are all nan:
+
+    >>> stats.percentileofscore([], 1)
+    nan
+    >>> stats.percentileofscore([np.nan, 1, 2], [1, 2], nan_policy='omit')
+    array([ 50., 100.])
+    >>> stats.percentileofscore([np.nan, np.nan], [1, 2], nan_policy='omit')
+    array([nan, nan])
+
+    The 'omit' policy does not make sense if there are nan's among the scores,
+    and so an error in that case.
     """
-    if np.isnan(score):
-        return np.nan
+
     a = np.asarray(a)
     n = len(a)
+    score = np.asarray(score)
+
+    # Nan treatment
+    cna, npa = _contains_nan(a, nan_policy, summary_check=False)
+    cns, nps = _contains_nan(score, nan_policy, summary_check=False)
+
+    if (cna or cns) and nan_policy == 'raise':
+        raise ValueError("The input contains nan values")
+
+    elif cns and nan_policy == "omit":
+        raise ValueError("The input scores contains nan values,"
+                         " which is incompatible with the 'omit' policy.")
+
+    # If a score is nan, then the output should be nan
+    if cns and nan_policy == "propagate":
+        score = ma.masked_where(np.isnan(score), score)
+
+    # Don't count nans
+    if cna and nan_policy == "omit":
+        a = ma.masked_where(np.isnan(a), a)
+        n = a.count()
+
+    # With `propagate`, an `a` with nans should always yield nan
+    if cna and nan_policy == "propagate":
+        n = 0
+
+    # Cannot compare to empty list ==> nan
     if n == 0:
-        return 100.0
-
-    if kind == 'rank':
-        left = np.count_nonzero(a < score)
-        right = np.count_nonzero(a <= score)
-        pct = (right + left + (1 if right > left else 0)) * 50.0/n
-        return pct
-    elif kind == 'strict':
-        return np.count_nonzero(a < score) / n * 100
-    elif kind == 'weak':
-        return np.count_nonzero(a <= score) / n * 100
-    elif kind == 'mean':
-        pct = (np.count_nonzero(a < score)
-               + np.count_nonzero(a <= score)) / n * 50
-        return pct
+        perct = np.full_like(score, np.nan, dtype=float)
+
     else:
-        raise ValueError("kind can only be 'rank', 'strict', 'weak' or 'mean'")
+        # Prepare broadcasting
+        score = score[..., None]
+        count = lambda x: np.count_nonzero(x, -1)
+
+        # Despite masking, Azure pipelines produces warnings
+        # when there are nans (which should not be processed)
+        with suppress_warnings() as sup:
+            sup.filter(RuntimeWarning,
+                       "invalid value encountered in less")
+            sup.filter(RuntimeWarning,
+                       "invalid value encountered in greater")
+
+            # Compute
```

Comment:
I'd suggest being more specific or omitting the comment.

Reply:
Please avoid resolving comments that have not been resolved. If you disagree, please reply; otherwise, please consider implementing the suggestion.

Reply:
I implemented the suggestion in 178d6fa . Not sure why it's not showing up here (at least by an "obsolete" tag), but I am not very familiar with the Github PR machinery yet.

Reply:
It doesn't look like it made it into that commit.

Reply:
Hmmm... I'm confused. The change is there. See the "files changed" tab, and ctrl-f for "Azure pipelines". I have expanded the comment.

Reply:
I think I see now. I think you thought I was referring to:
![image](https://user-images.githubusercontent.com/6570539/116927502-9984bf00-ac10-11eb-8347-82d093d90211.png)

This comment was attached to the line:
![image](https://user-images.githubusercontent.com/6570539/116927545-a86b7180-ac10-11eb-8b4a-c143504d630c.png)

That's the comment I was referring to.

Reply:
Ahhh.... sorry, I thought the lines it was pointing/attached to was a side-effect of other changes I had made. I have made the comment a bit more descriptive in the latest commit. If it's still not quite good, I think I will just remove the comment.